### DISCUSSION TITLE
Specific class names in input.css not compiling?

### DISCUSSION THREAD

**Discussion by robMarcoPolo** [tailwindlabs/tailwindcss#12746] [Help] [answered]
This is a curious one.  

I'm trying to add some specific @apply rules to a specific class used by Editor.js (".ce-block__content").

The specific rule I'm trying to apply is as follows:

input.css:
```
/* Editor.js */

 .ce-block__content {
        @apply prose;
    }

/* End Custom CSS */
```

However, when the CSS is built (using Tailwind CLI), the rule(s) aren't added to the CSS file, while the comments are included:

output.css:
```
/* Editor.js */

/* End Custom CSS */
```

Now, the odd thing is when I change the class to "test", it seems to compile successfully:

input.css
```
/* Editor.js */

    .test {
        @apply prose;
    }

/* End Custom CSS */
```

output.css
```
/* Editor.js */

.test {
  color: var(--tw-prose-body);
  max-width: 65ch;
}

.test :where(p):not(:where([class~="not-prose"],[class~="not-prose"] *)) {
  margin-top: 1.25em;
  margin-bottom: 1.25em;
}

...
```

This seems to imply there's something fundamentally wrong with the "ce-block__content" class name used?  But I can't tell what.  There's no duplication of that class name within the input file, so I'm at a bit of a loss!

**Comment by wongjn** [comment, marked as answer]
Do you have the `.ce-block__content` rule in a `@layer base/components/utilities` rule? If it is, then it would be subject to the same JIT rules as all the other class candidates – Tailwind would not compile the class unless it sees that string in files that it scans. Perhaps `.test` works because you *do* have `test` as a string somewhere in files that Tailwind scans – `test` would be a fairly common word to use code.

You could consider declaring the `.ce-block__content` rule outside any `@layer base/components/utilities` rule or you could try [safelisting it](https://tailwindcss.com/docs/content-configuration#safelisting-classes).

**Reply by robMarcoPolo** [reply]
Ah, that seems to have done the trick - thank you!  

I think I need to brush up on why/when you include rules within the @layer base/components/utilities rule, and when you can include it outside.
